Cython/Includes/cpython/datetime.pxd

```diff
@@ -161,11 +288,42 @@ cdef inline object datetime_new(int year, int month, int day, int hour, int minu
 cdef inline object timedelta_new(int days, int seconds, int useconds):
     return PyDateTimeAPI.Delta_FromDelta(days, seconds, useconds, 1, PyDateTimeAPI.DeltaType)
 
+# Create timedelta object using DateTime CAPI factory function.
+cdef inline object timezone_new(object offset, str name=None):
+    if PY_VERSION_HEX < 0x030700b1:
+        raise RuntimeError('Time zones are not available from the C-API.')
+    return PyTimeZone_FromOffsetAndName(offset, name)
+
+# Create datetime object using PEP 495 constructor.
+cdef inline object datetime_and_fold_new(
+        int year, int month, int day, int hour, int minute, int second, int microsecond, object tz, int fold
+    ):
+    return __Pyx_DateTime_DateTimeWithFold(year, month, day, hour, minute, second, microsecond, tz, fold)
+
+# Create time object using PEP 495 constructor.
+cdef inline object time_and_fold_new(int hour, int minute, int second, int microsecond, object tz, int fold):
+    return __Pyx_DateTime_TimeWithFold(hour, minute, second, microsecond, tz, fold)
```

Comment:
I don't think we need these as new functions. Just add `fold=0` as a default argument to `datetime_new()` and `time_new()`, and let them call the new macros as implementation.